packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1876,6 +1592,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 this.updateDocumentDirtyState(false);
             }
 
+            if (message.metadata?.compressed) {
+                this.logger.sendTelemetryEvent({
+                    eventName: "opToDecompress",
+                    message: JSON.stringify(message),
+                });
+
+                const contents = IsoBuffer.from(message.contents, "base64");
```

Comment:
I do not know if it's actionable (or rather - if we want to tie this work together, as it will slow down this work substantially), but worth pointing the best outcome here:
1. Runtime operates with binary content (compressed or not).
2. Driver does base64 encoding / decoding on as needed bases.

The reason why this flow is the best - we are more likely than not to be in situation where one storage provider will implement binary payloads sooner than another, or maybe some storage / driver will never do it at all (we might not care for such things as memorylicious, tinylicious).

Reason it complicates things:
1. Splitting the code across layers will require careful consideration, as these layers ship to production at various times. So, we will need to carefully take care of interesting permutations of "old" and "new" code at different layers.
2. Ideally, we change type system to reflect with what we are working. See https://dev.azure.com/fluidframework/internal/_workitems/edit/954, where I mention that protocol & loader should work with string, but runtime should work with object/any. You change basically asks for string | binary at Loader / driver / core protocol layers, and even further split (a particular driver should define their own type).
    - I'd definitely not do this work now and add to this task.